posts/inside-rust/2022-07-29-compiler-team-2022-midyear-report.md

```diff
@@ -0,0 +1,800 @@
+we think the most important parts of the problem are now (or very soon to be) solved, and additional parts of the solution will be available in the next six months.
+
+**How it started:** At start of 2022, we had a plan, some of which had been completed, but more work remained to be done. Furthermore, that plan was not sufficient and was in need of revision before it could actually solve the problem.
+
+**How it's going:** we understand the problem better than we did at the start of the year,
+we have a high-level end-to-end plan to solve the problem,
+we have milestones that represent individual units of work for the near term that each make progress towards a solution
+we have Rust contributors who have agreed to help with the units of work that we have identified.
+Further more, we have completed some of our milestones; we think we have implemented a solution to the problem, but most of our work has not yet landed in hands of Rust users
+
+**Details:** At the start of the year, we opened [PR #92268](https://github.com/rust-lang/rust/pull/92268), *Initial Implementation of Transmutability Trait*, which aimed to provide the basic functionality of a trait implemented for any two types transmutable into each other (as defined by [MCP #411](https://github.com/rust-lang/compiler-team/issues/411)). This PR required additional testing and polish before it would be ready to merge, but progress unfortunately stalled in the spring.
+
+With the mentoring provided by [@oli-obk](https://github.com/oli-obk) and an influx of interest and help from [@m1el](https://github.com/m1el), progress resumed this summer; notably:
+- A significant effort in testing revealed flaws in the initial implementation approach. Fortunately, we quickly [discovered](https://rust-lang.zulipchat.com/#narrow/stream/216762-project-safe-transmute/topic/Implementation/near/288584316) and implemented an alternative (and arguably simpler) implementation strategy!
+- The `rustc_transmute` crate now only *optionally* depends on other `rustc_*` dependencies, allowing contributors to edit, build, and test the core implementation using the familiar `cargo` commands, rather than building the entire compiler.
+
+[PR #92268](https://github.com/rust-lang/rust/pull/92268) is now undergoing the final polish required for it to be merged, and [near-future units of follow-up work](https://rust-lang.zulipchat.com/#narrow/stream/216762-project-safe-transmute/topic/Implementation/near/290258987) have been identified.
+
+**Regarding new contributors:** An influx of interest and help from [@m1el](https://github.com/m1el) ([Igor null](https://rust-lang.zulipchat.com/#narrow/stream/216762-project-safe-transmute/sender/498326-user498326) on Zulip) jolted Project Safe Transmute out of its doldrums. Additionally, [@joshlf](https://github.com/joshlf), an early collaborator on Project Safe Transmute, anticipates he will soon be able to rejoin the implementation effort.
```

Comment:
Nit: My understanding is that m1el/Igor Null are the same person so I'm removing that part of the sentance.

```suggestion
**Regarding new contributors:** An influx of interest and help from @m1el jolted Project Safe Transmute out of its doldrums. Additionally, @joshlf, an early collaborator on Project Safe Transmute, anticipates he will soon be able to rejoin the implementation effort.
```